File: JavaScript/Array/L_convert1DArrTo2DArr.js
var construct2DArray = function (original, m, n) {
    let result = new Array(m).fill().map(() => []);
    switch (m * n === original.length ? 1 : 0) {
        case 1:
            let i = 0;
            while (i < m) {
                result[i] = original.slice(i * n, i * n + n);
                i++;
            }
            break;
        default:
            return [];
    }
    return result;
};

File: cpp/Array/L_convert1DArrTo2DArr.cpp


vector<vector<int>> construct2DArray(vector<int> &original, int m, int n)
{
    if (m * n != original.size())
    {
        return {};
    }

    std::vector<std::vector<int>> result(m, std::vector<int>(n));

    for (int i = 0; i < m; ++i)
    {
        result[i] = std::vector<int>(original.begin() + i * n, original.begin() + i * n + n);
    }

    return result;
}